src/dataflow/src/render/debezium.rs

```diff
@@ -118,6 +124,273 @@ where
+                                                None => output.session(&binary_cap).give((
+                                                    Err(err),
+                                                    *binary_cap.time(),
+                                                    1,
+                                                )),
+                                            }
+                                            continue;
+                                        }
+                                    }
+                                }
+                                None => true,
+                            };
+
+                            if should_use {
+                                let tx_cap_entry = match tx_cap_map.entry(tx_id.to_string()) {
+                                    Entry::Occupied(e) => e,
+                                    Entry::Vacant(_) => panic!("Must have cap if using record"),
+                                };
+
+                                let mut session = output.session(&**tx_cap_entry.get());
+                                match value.iter().nth(before_idx).unwrap() {
+                                    Datum::List(l) => {
+                                        session.give((Ok(Row::pack(&l)), tx_time, -1));
+                                    }
+                                    Datum::Null => {}
+                                    d => {
+                                        panic!("type error: expected record, found {:?}", d)
+                                    }
+                                }
+                                match value.iter().nth(after_idx).unwrap() {
+                                    Datum::List(l) => {
+                                        session.give((Ok(Row::pack(&l)), tx_time, 1));
+                                    }
+                                    Datum::Null => {}
+                                    d => {
+                                        panic!("type error: expected record, found {:?}", d)
+                                    }
+                                }
+                                match tx_event_count.get_mut(tx_id) {
+                                    Some(count) => {
+                                        *count -= 1;
+                                        if *count == 0 {
+                                            let (_, tx_cap) = tx_cap_entry.remove_entry();
+                                            binary_cap.downgrade(&tx_cap.time());
+                                        }
+                                    }
+                                    None => panic!("need event count for tx_id {:?}", tx_id),
+                                }
+                            }
+                        }
+
+                        if !data_buffer.is_empty() || !tx_cap_map.is_empty() {
+                            activator.activate_after(Duration::from_secs(1));
```

Comment:
We don't need to manually activate the operator here because we won't make progress anyway by reprocessing the state. The only way to make forward progress is for new data to arrive, which will cause timely to schedule us anyway.